worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,47 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e){
+    haveFailuresChanged(e.document);
+}
+
+function haveFailuresChanged(document) {
+    if (document.getFailures().isChanged()) {
+
+        var listOfFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfFailures.add(failure);
+        });
+
+        document.getFailures().reset();
+
+        var listOfOriginalFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfOriginalFailures.add(failure);
+        });
+
+        var newListOfFailures = new java.util.ArrayList();
+        for each (var f in listOfFailures) {
+            if (!listOfOriginalFailures.contains(f)) {
+                var message = {
+                    id: f.getFailureId(),
+                    stack: f.getFailureStack(),
+                    description: {
```

Comment:
The `description` level doesn't give us anything.  Drop it and raise it's fields up to this level.